netjsonconfig/backends/raspbian/templates/hostapd.jinja2

```diff
@@ -0,0 +1,26 @@
+{% for wireless in data.wireless %}
+    # config: /etc/hostapd/hostapd.conf
+
+    interface={{ wireless.get('ifname') }}
+    driver=nl80211
+    {% if wireless.get('protocol') == 'a' or 'b' or 'g' %}
+    hw_mode={{ wireless.get('hwmode')}}
+    {% endif %}
+    channel={{ wireless.get('channel')}}
+    {% if wireless.get('protocol') == '80211n' %}
+    ieee80211n=1
+    {% endif %}
+    {% if wireless.get('protocol') == '80211ac' %}
+    ieee80211ac=1
+    {% endif %}
+    ssid={{ wireless.get('ssid') }}
+    {% if wireless.get('encryption') != {} %}
+        auth_algs={{ wireless.get('encryption').get('auth_algs') }}
+        wpa={{ wireless.get('encryption').get('wpa') }}
+        wpa_key_mgmt={{ wireless.get('encryption').get('wpa_key_mgmt') }}
+        wpa_passphrase={{ wireless.get('encryption').get('wpa_passphrase') }}
+        {% if wireless.get('encryption', None).get('wpa_pairwise') != 'AUTO' %}
```

Comment:
The following code:

```
wireless.get('encryption', None).get('wpa_pairwise')
```

is broken because if encryption is not present, `wireless['encryption']` will be `None` and then you will be trying to call the `get` method on a `None` object, which will raise an exception.

But if you apply my previous suggestion, this code block will be executed only if `wireless.encryption` is not empty or `None`, hence the default value in the get method is not necessary and you may write just:

```
{% if  wireless.encryption.wpa_pairwise != 'AUTO' %}
```